### PR TITLE
Add 5250 keyboard reset button

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

Adds a button to the terminal view that is displayed when the user attempts to type in an area where they cannot in 5250.

Not many users know the tn5250 shortcut (Control+R) to do a reset, so I added a button to the UI for it.

### How to test this PR

Examples:

1. Connect and launch a 5250
2. Sign in
3. Attempt to write outside an input field
4. See the button appear in the terminal tab
5. Click the button and see the cursor position reset

### Checklist

* [x] have tested my change
* [ ] have created one or more test cases
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/codefori/vscode-ibmi/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by worksofliam** [comment]
Fixed the wording: 

image

File: src/api/Terminal.ts [hunk 1]
```diff
@@ -31,6 +32,35 @@ export namespace Terminal {
...
+          const ifsPath = (await content.isDirectory(ifsNode.path)) ? ifsNode.path : path.dirname(ifsNode.path);
+          const terminal = await selectAndOpen(instance, TerminalType.PASE);
+          terminal?.sendText(`cd ${ifsPath}`);
+        }
+      }),
+
+      vscode.commands.registerCommand(`code-for-ibmi.term5250.systemAttention`, () => {
+        const term = vscode.window.activeTerminal;
+        if (term) {
+          term.sendText(Buffer.from([1, 9]).toString(), false);
```
Comment (chrjorgensen) [inline src/api/Terminal.ts:57]: Where does `Buffer` come from?

File: package.json [hunk 2]
```diff
@@ -1663,6 +1663,13 @@
 				"category": "IBM i",
 				"icon": "$(search-fuzzy)",
 				"enablement": "code-for-ibmi:connected"
+			},
+			{
+				"command": "code-for-ibmi.term5250.systemAttention",
+				"title": "5250 Attention Request",
```
Comment (chrjorgensen) [inline package.json:1669]:
```suggestion
				"title": "Reset keyboard",
```

File: src/api/Terminal.ts [hunk 3]
```diff
@@ -71,13 +101,20 @@ export namespace Terminal {
     }
   }
 
+  const HALT_START = `[24;2H(B[0;1m[37m[40m`;
```
Comment (chrjorgensen) [inline src/api/Terminal.ts:104]: Why does these lines look like this in VS Code?

billede

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
@worksofliam The code works (I don't know why - see my comment on `Buffer`), but the button is only available on the login screen, not any subsequent panels.

I also think there are still some misunderstandings / wrong wordings here:

- The Attn (attention) key _does_ reset the keyboard when locked - but it does more than that: It calls the program set by the SETATNPGM command in the job.
- What you're after is the Reset button (Ctrl-R) - it resets the keyboard and nothing more.
- The title of the VS Code command should be changed as commented.
- The names used for VS Code commands and TypeScript functions should reflect this.
- The title of this PR should reflect this.

👍

File: src/api/Terminal.ts (hunk 1 above)
Comment (worksofliam) [reply src/api/Terminal.ts:57]: It's a Node.js built-in! https://nodejs.org/api/buffer.html

Comment (chrjorgensen) [reply src/api/Terminal.ts:57]: Ah - so you're sending bytes 0x01 and 0x09 to the terminal, or how should I interpret it? I'm not jetlagged but I don't get it... it's not obvious to me! 😆

Comment (chrjorgensen) [reply src/api/Terminal.ts:57]: Maybe some constants to explain what these codes do would help on my understanding... 🙄 😄

File: src/api/Terminal.ts (hunk 3 above)
Comment (worksofliam) [reply src/api/Terminal.ts:104]: Because these are bytes which don't have a character representation.

File: src/api/Terminal.ts (hunk 1 above)
Comment (worksofliam) [reply src/api/Terminal.ts]: Done!

**Review by chrjorgensen — CHANGES_REQUESTED** [review]
@worksofliam Almost there... 😉

It is named correctly and works on all panels. 👍 
Only remaining issues are
- the Halt status is not reset if user press Ctrl-R and unlocks the keyboard
- you test for `II` - so pressing `Enter` on any panel containing `II` wil cause the Halt status:
billede
Maybe better to check for `5250(5blanks)X II`?
billede

**Review by chrjorgensen — APPROVED** [review]
@worksofliam Approved and merged. 👍

**Pull request merged**